osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1419,206 +1446,271 @@ namespace renderkit {
+		bool hasEyes = hasLeftEyePose() && hasRightEyePose();
+		if (!hasEyes)
+		{
+			if (whichEye % 2 == 0) {
+				// Left eye
+				q_headFromRotatedEye.xyz[Q_X] -= params.IPDMeters / 2;
+			}
+			else {
+				// Right eye
+				q_headFromRotatedEye.xyz[Q_X] += params.IPDMeters / 2;
+			}
+		}
+		else
+		{
+			if (whichEye % 2 == 0) {
+				// Left eye
+				if (params.roomFromLeftEyeReplace != nullptr) {
+					q_from_OSVR(q_roomFromEye, *params.roomFromLeftEyeReplace);
+				}
+				else
+				{
+					/// Use the state interface to read the most-recent
+					/// location of the left eye.  It will have been updated
+					/// by the most-recent call to update() on the context.
+					/// DO NOT update the client here, so that we're using the
+					/// same state for all eyes.
+					if (!m_leftEyePoseCache || !m_leftEyePoseCache->getLastReport(timestamp, m_roomFromLeftEye)) {
+						// This it not an error -- they may have put in an invalid
+						// state name for the left eye; we just ignore that case.
+					}
+					q_headFromRotatedEye.xyz[Q_X] -= m_roomFromLeftEye.translation.data[0];
+					q_headFromRotatedEye.xyz[Q_Y] -= m_roomFromLeftEye.translation.data[1];
+					q_headFromRotatedEye.xyz[Q_Z] -= m_roomFromLeftEye.translation.data[2];
+
+				}
+			}
+			else {
+				/// Use the state interface to read the most-recent
+				/// location of the right eye.  It will have been updated
+				/// by the most-recent call to update() on the context.
+				/// DO NOT update the client here, so that we're using the
+				/// same state for all eyes.
+				if (!m_rightEyePoseCache || !m_rightEyePoseCache->getLastReport(timestamp, m_roomFromRightEye)) {
+					// This it not an error -- they may have put in an invalid
+					// state name for the right eye; we just ignore that case.
+				}
+				q_headFromRotatedEye.xyz[Q_X] -= m_roomFromRightEye.translation.data[0];
+				q_headFromRotatedEye.xyz[Q_Y] -= m_roomFromRightEye.translation.data[1];
+				q_headFromRotatedEye.xyz[Q_Z] -= m_roomFromRightEye.translation.data[2];
+			}
+		}
+
+		q_xyz_quat_type q_headFromEye;
+		q_xyz_quat_compose(&q_headFromEye, &q_headFromRotatedEye,
+			&q_rotatedEyeFromEye);
+
+		/// Include the impact of RenderParams.headFromRoom
+		/// (which will override m_headFromRoom) or of m_headFromRoom.
+		q_xyz_quat_type q_roomFromHead;
+		if (params.roomFromHeadReplace != nullptr) {
+			/// Use the params.m_headFromRoom as our transform
+			q_from_OSVR(q_roomFromHead, *params.roomFromHeadReplace);
+		}
+		else {
+			/// Use the state interface to read the most-recent
+			/// location of the head.  It will have been updated
+			/// by the most-recent call to update() on the context.
+			/// DO NOT update the client here, so that we're using the
+			/// same state for all eyes.
+			OSVR_TimeValue timestamp;
+			if (!m_headPoseCache || !m_headPoseCache->getLastReport(timestamp, m_roomFromHead)) {
+				// This it not an error -- they may have put in an invalid
```

Comment:
It may not be an error but are we doing the right fallback here? Is m_roomFromHead valid data if this if statement falls through to the code below?